test/ApiFirstMediatR.Generator.Tests/SourceGenerators/ApiSpecDiagnosticTests.cs

```diff
@@ -14,25 +14,22 @@ public void MissingAPISpecFile_ThrowsDiagnostic()
             .RunGeneratorsAndUpdateCompilation(inputCompilation, out var outputCompilation,
                 out var diagnostics);
 
-        diagnostics.Should().ContainSingle()
-            .Which.Id.Should().Be(DiagnosticIdentifiers.ApiSpecFileNotFound);
+        diagnostics.Should().Contain(o => o.Id == DiagnosticIdentifiers.ApiSpecFileNotFound);
```

Comment:
As discussed, these tests can be reverted now that the multiple iteration issue is fixed